Sources/Sentry/SentrySubClassFinder.m

```diff
@@ -24,115 +25,73 @@ - (instancetype)initWithDispatchQueue:(SentryDispatchQueueWrapper *)dispatchQueu
+#if TEST
         // Only for testing. We want to know in tests if the code iterated over the classes, because
         // iterating in edge cases could lead to crashses. Ideally, we would wrap
         // class_getSuperclass in the SentryObjCRuntimeWrapper and count its invocations. As
         // class_getSuperclass is called in a tight loop doing so would slow down the code
         // significantly. This is pragmatic workaround to find out in tests if the code iterated
         // over the classes.
         [self.objcRuntimeWrapper countIterateClasses];
+#endif
 
         // Storing the actual classes in an NSArray would call initializer of the class, which we
         // must avoid as we are on a background thread here and dealing with UIViewControllers,
-        // which assume they are running on the main thread. Therefore, we store the indexes instead
-        // so we can search for the subclasses on a background thread.
-        // We can't use NSObject:isSubclassOfClass as not all classes in the runtime in classes
-        // inherit from NSObject and a call to isSubclassOfClass would call the initializer of the
-        // class, which we can't allow because of the problem with UIViewControllers mentioned
-        // above.
-
-        NSMutableArray<NSNumber *> *indexesToSwizzle = [NSMutableArray new];
-        for (NSInteger i = 0; i < numClasses; i++) {
-            Class superClass = classes[i];
-
-            // Don't add the parent class to list of sublcasses
-            if (superClass == parentClass) {
-                continue;
-            }
-
-            // Using a do while loop, like pointed out in Cocoa with Love
-            // (https://www.cocoawithlove.com/2010/01/getting-subclasses-of-objective-c-class.html)
-            // can lead to EXC_I386_GPFLT which, stands for General Protection Fault and means we
-            // are doing something we shouldn't do. It's safer to use a regular while loop to check
-            // if superClass is valid.
-            while (superClass && superClass != parentClass) {
-                superClass = class_getSuperclass(superClass);
-            }
-
-            if (superClass) {
-                [indexesToSwizzle addObject:@(i)];
+        // which assume they are running on the main thread. Therefore, we store the class name
+        // instead so we can search for the subclasses on a background thread. We can't use
+        // NSObject:isSubclassOfClass as not all classes in the runtime in classes inherit from
+        // NSObject and a call to isSubclassOfClass would call the initializer of the class, which
+        // we can't allow because of the problem with UIViewControllers mentioned above.
+        NSMutableArray<NSString *> *classesToSwizzle = [NSMutableArray new];
+        for (int i = 0; i < count; i++) {
+            NSString *className = [NSString stringWithUTF8String:classes[i]];
+            if ([className hasSuffix:@"ViewController"]) {
```

Comment:
`m`: Shouldn't we maybe go with `containsString` as we anyways check if the class is a sub-class of UIViewController?

Reply:
Sure, don't see why not.